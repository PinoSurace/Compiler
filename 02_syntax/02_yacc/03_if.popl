IF var THEN
  Print("ok");
ENDIF;

IF ( var + 7 ) < -var*9 THEN Do; ELSE Something; ENDIF;

[ this does not nest correctly ]
IF var THEN
  Print;
  IF var * -1 THEN
    Do;
  ELSE
    Something;
ELSE
  Print("error");
ENDIF;
