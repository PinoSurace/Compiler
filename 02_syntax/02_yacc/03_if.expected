atom( var )
factor
term
expr
atom( ok )
factor
term
expr
function_call
statement_seq
if_statement
statement_seq
atom( var )
factor
term
atom( 7 )
factor
term
expr
atom
factor
term
atom( var )
factor
atom( 9 )
factor
term
term
expr
function_call
statement_seq
function_call
statement_seq
if_statement
statement_seq
atom( var )
factor
term
expr
function_call
statement_seq
atom( var )
factor
atom( 1 )
factor
term
term
expr
function_call
statement_seq
function_call
statement_seq
14:Syntax Error (token: 'ELSE')
